frontend-v2/src/features/data/MapHeaders.tsx

```diff
@@ -39,55 +49,81 @@ const MapHeaders: FC<IMapHeaders> = ({
-                        {header}
-                      </MenuItem>
-                    )),
-                  ])}
-                </Select>
-              </FormControl>
-            </TableCell>
-          ))}
-        </TableRow>
-      </TableHead>
-      <TableBody>
-        {data.map((row, index) => (
-          <TableRow key={index}>
-            {fields.map((field, index) => (
-              <TableCell key={index}>{row[field]}</TableCell>
+    <>
+      <TableContainer
+        sx={{
+          overflow: "auto",
+          maxHeight: "inherit",
+        }}
+      >
+        <Table size="small">
+          <TableHead>
+            <TableRow>
+              {fields.map((field, index) => (
+                <TableCell key={index}>
+                  <Typography
+                    variant="h6"
+                    component="div"
+                    sx={{ flexGrow: 1, marginBottom: 1 }}
+                    align="center"
+                  >
+                    {field}
+                  </Typography>
+                  <FormControl fullWidth>
+                    <InputLabel id={`select-${index}-label`}>
+                      Column Type
+                    </InputLabel>
+                    <Select
+                      labelId={`select-${index}-label`}
+                      id={`select-${index}`}
+                      value={normalisedFields.get(field)}
+                      label="Column Type"
+                      onChange={handleFieldChange(field)}
+                    >
+                      {Object.entries(groupedHeaders).map(
+                        ([group, headers]) => [
+                          <ListSubheader key={group}>{group}</ListSubheader>,
+                          ...headers.map((header) => (
+                            <MenuItem key={header} value={header}>
+                              {header}
+                            </MenuItem>
+                          )),
+                        ],
+                      )}
+                    </Select>
+                  </FormControl>
+                </TableCell>
+              ))}
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {visibleRows.map((row, index) => (
+              <TableRow key={index}>
+                {fields.map((field, index) => (
+                  <TableCell key={index}>{row[field]}</TableCell>
+                ))}
+              </TableRow>
             ))}
-          </TableRow>
-        ))}
-      </TableBody>
-    </Table>
+          </TableBody>
+        </Table>
+      </TableContainer>
+      <TablePagination
+        rowsPerPageOptions={rowsPerPageOptions}
+        component="div"
+        count={data?.length || 0}
+        rowsPerPage={rowsPerPage}
+        page={page}
+        onPageChange={handlePageChange}
+        onRowsPerPageChange={handleChangeRowsPerPage}
+        sx={{ display: 'flex', justifyContent: 'center', flexShrink: '0'}}
+      />
```

Comment:
This works but `TablePagination` should be placed inside a `TableFooter`, inside `TableContainer`. See the [MUI Table examples](https://mui.com/material-ui/react-table/).